### DISCUSSION TITLE
How do I create an `AssetSelection` of my dbt assets using dbt tags?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18282] [Q&A] [answered]
Hello! I am strking out on resources for this. I am looking to convert a dbt_asset definition to a job. The trick is, I want the job to match the dbt tag definition in the dbt_asset construction. I can't seem to find the AssetSelection call that will translate that properly. I have more dbt assets in my `group` that I originally built out that I want to be called in a second job. I really don't want to manage this all through both dbt tags and groups matching 1:1.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C04CW71AGBW/p1701100395917079?thread_ts=1701100395.917079&cid=C04CW71AGBW).*

**Comment by rexledesma** [comment, marked as answer]
You can build an [`AssetSelection`](https://docs.dagster.io/_apidocs/assets#dagster.AssetSelection) of dbt assets using [`build_dbt_asset_selection`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.build_dbt_asset_selection). This creates a [`DbtManifestAssetSelection`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtManifestAssetSelection) that defines a selection of assets from a dbt manifest and a dbt selection string.

References:
- https://docs.dagster.io/integrations/dbt/reference#scheduling-jobs-that-contain-dbt-assets-and-non-dbt-assets

**Reply by pramabhi** [reply]
Hi @rexledesma 
foo_selection = build_dbt_asset_selection([dbt_assets], dbt_select="tag:foo")
The line above is giving the following error.
`lib/python3.10/dist-packages/dagster_dbt/asset_utils.py", line 305, in get_manifest_and_translator_from_dbt_assets`
`metadata_by_key = dbt_assets_def.metadata_by_key or {}`
`AttributeError: 'list' object has no attribute 'metadata_by_key'`

**Reply by AlbertoBSJR** [reply]
I´m facing the same issue, how Can I solve it?
